Logger/BasePerson.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Logger;
+
+public abstract record class BasePerson : BaseEntity
+{
+    // Implicit implementation so that we can create a class-specific Name implementation
+    public override string Name
+    {
+        get
+        {
+            if (string.IsNullOrWhiteSpace(FullName.MiddleName))
+                return FullName.FirstName + " " + FullName.LastName;
+            else
+                return FullName.FirstName + " " + FullName.MiddleName + " " + FullName.LastName;

```

Comment:
string.Format is a good way to make these return lines more readable, that way you don't have this string arithmetic going on for spaces and piecing it together.
```suggestion
                return string.Format($"{FullName.FirstName} {FullName.MiddleName} {FullName.LastName}");

```
(This is for this line as an example)